### DISCUSSION TITLE
Commented rules in math.rs test

### DISCUSSION THREAD

**Discussion by bedeio** [egraphs-good/egg#181] [Q&A] [answered]
Is there any reason why the following two rules are commented out in the `math.rs` test?
```
    // rw!("canon-sub"; "(+ ?a (* -1 ?b))"   => "(- ?a ?b)"),
    // rw!("canon-div"; "(* ?a (pow ?b -1))" => "(/ ?a ?b)" if is_not_zero("?b")),
```

Do they create an infinite loop, as described [here](https://github.com/egraphs-good/egg/discussions/60)?

**Comment by mwillsey** [comment, marked as answer]
We found that those rules were not necessary to prove any of the things we
wanted to prove. As far as I can remember, there's nothing wrong with
including them, although in general more rules = slower.
